sky/clouds/service_catalog/__init__.py

```diff
@@ -107,11 +107,17 @@ def get_region_zones_for_instance_type(
 
 def get_hourly_cost(instance_type: str,
                     region: Optional[str],
+                    zone: Optional[str],
                     use_spot: bool,
                     clouds: CloudFilter = None):
-    """Returns the cost, or the cheapest cost among all zones for spot."""
+    """Returns the price of a VM instance in the given region and zone.
+
+    If region is None, return the cheapest price among all regions.
+    If zone is None, return the cheapest price among all the zones in the given
```

Comment:
Good point. Added more detailed comments.

BTW, seems like #1014 allowed region to be None when cloud and zone are specified (e.g., `Resources(region=None, zone='us-west1-a', cloud=sky.GCP())`). Originally I disallowed this because AWS and GCP have very similar zone names; e.g., one of the AWS zones is `us-west-1a` whereas GCP has `us-west1-a`. But now that cloud is always specified with the zone, we'll not have such ambiguity. So I fixed the PR to allow region to be None.